### ISSUE TITLE
Source of zwave data?

### ISSUE THREAD

**Issue by radical-squared** [scaarup/hass-danfoss#1]
Looks very interesting. Thanks for sharing. How do you get zwave input from Danfoss?

**Comment by scaarup** [comment]
That can be accomplished with a passive device listening on the z-wave network. Could be  https://www.suphammer.net/suphacap or an rtl-sdr dongle: https://www.rtl-sdr.com/buy-rtl-sdr-dvb-t-dongles/

**Comment by radical-squared** [comment]
Thanks for the hint. Quick question before ordering one - does it read Danfoss proprietary z-wave protocol?

**Comment by scaarup** [comment]
You will see unencrypted traffic in the form of hexadecimals. This is where my decoding script will come in handy.

**Comment by radical-squared** [comment]
This is great! Just ordered the rtl-sdr dongle :) 

I have two zwave networks at home - Danfoss' proprietary zwave with about 20 of their smart radiator valves and thermostats and a standard zwave with another 20+ devices (sensors and switches). In your script's source code I haven't noticed a network key. Is it able to differentiate danfoss and non-danfoss devices?

**Comment by scaarup** [comment]
I am in the same situation as you :). You will quickly figure out which homeid is your danfoss.
I have another experimental script, which I can expose here. Maybe you can use that for inspiration.

**Comment by scaarup** [comment]
Take a look at the docker branch: https://github.com/scaarup/hass-danfoss/tree/docker/docker. Was created to run in a container with an rtl-sdr dongle, and use the decode.sh script to parse and push to Home Assistant.

**Comment by radical-squared** [comment]
OMG, that makes it much easier :) Thanks for sharing! Any chance to use to dongle to inject packets to the network to set the radiator valves?

**Comment by scaarup** [comment]
Yes I would think so. But then you need to invest more money in the project and perhaps buy a HackRF: https://greatscottgadgets.com/hackrf/one/

**Comment by radical-squared** [comment]
Not that expensive considering how much money I have spent on danfoss already :) Have you got one?

**Comment by scaarup** [comment]
Yeah, that's right :). No I havn't got one. It is not really my goal to take over the heating control.

**Comment by radical-squared** [comment]
I see. I got the rtl_sdr dongle yesterday and got it to work, but the data it provides seems incomplete and confusing. I ordered suphacap (with custom danfoss firmware) earlier today and hope it works better. 

Although bash is not my mother tongue, your scripts are mostly self explanatory. Got to admit the regexp and calcs are quite overwhelming though :) One thing that is not clear to me is coloring of ids (e.g. colorizeIds in the danfoss.sh script). What is it intended for?

**Comment by MarcoSuurlant** [comment]
> 
> 
> I see. I got the rtl_sdr dongle yesterday and got it to work, but the data it provides seems incomplete and confusing. I ordered suphacap (with custom danfoss firmware) earlier today and hope it works better.
> 
> Although bash is not my mother tongue, your scripts are mostly self explanatory. Got to admit the regexp and calcs are quite overwhelming though :) One thing that is not clear to me is coloring of ids (e.g. colorizeIds in the danfoss.sh script). What is it intended for?

Hi how far are you now I also want to look in this.

**Comment by scaarup** [comment]
I am in the progress of switching to Danfoss Ally thermostats...

**Comment by hoegge** [comment]
>  I ordered suphacap (with custom danfoss firmware) earlier today and hope it works better.

Hi What does "with custom Danfoss firmware" mean? 
I have a hard time figuring out if anybody have really decoded the communication with the Danfoss Link / Living Connect thermostats or not.

**Comment by scaarup** [comment]
A firmware that makes it easy for us to decode the traffic. I managed to decode information like:

- set temperature
- actual temperature
- Battery level

So yes, I would say someone has managed to decode the comms.

**Comment by hoegge** [comment]
> 

That is amazing. Thanks for the answer. What HW device have you used for this?

**Comment by scaarup** [comment]
> > 
> 
> That is amazing. Thanks for the answer. What HW device have you used for this?

https://www.suphammer.net/suphacap or an rtl-sdr dongle: https://www.rtl-sdr.com/buy-rtl-sdr-dvb-t-dongles/

**Comment by hoegge** [comment]
> 

Yes thanks saw that -was just not aware if you had tried both or only one. Just want to improve my chance of success. Are they equally "easy" to use for this purpose?

**Comment by scaarup** [comment]
I would say the suphacap is best, since I have this problem with using my rtl-sdr: https://github.com/baol/waving-z/issues/8

**Comment by MarcoSuurlant** [comment]
Is it possible to make manual how to implement in HA?

**Comment by MarcoSuurlant** [comment]
> > 
> 
> Yes thanks saw that -was just not aware if you had tried both or only one. Just want to improve my chance of success. Are they equally "easy" to use for this purpose?

Hi, I also received the suphacap today, how far have you got with the code?
Could you help me to get it running in home assistant?
